src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -92,8 +97,17 @@ public class DefaultHtmlGenerator : IHtmlGenerator
 
             // Underscores are fine characters in id's.
             IdAttributeDotReplacement = optionsAccessor.Value.HtmlHelperOptions.IdAttributeDotReplacement;
+
+            MaxLengthAttributeRenderingEnabled = optionsAccessor.Value.MaxLengthAttributeRenderingEnabled;
         }
 
+        /// <summary> 
+        /// Gets or sets a value that indicates whether the maxlength attribute should be rendered for compatible HTML input elements, 
+        /// when they're bound to models marked with either 
+        /// <see cref="StringLengthAttribute"/> or <see cref="MaxLengthAttribute"/> attributes. 
+        /// </summary>
+        public bool MaxLengthAttributeRenderingEnabled { get; }
```

Comment:
`protected`

Reply:
Or private? Users could always get to the option if they super cared about it